app/main/forms/suppliers.py

```diff
@@ -34,11 +34,37 @@ class EditContactInformationForm(Form):
         DataRequired(message="You must provide an email address"),
     ])
     phoneNumber = StripWhitespaceStringField('Contact phone number')
+    # TODO: remove these fields when address is removed from the supplier details edit page
     address1 = StripWhitespaceStringField('Building and street')
     city = StripWhitespaceStringField('Town or city')
     postcode = StripWhitespaceStringField('Postcode')
 
 
+class EditRegisteredAddressForm(Form):
+    id = IntegerField()
+    address1 = StripWhitespaceStringField('Building and street', validators=[
+        DataRequired(message="You need to enter the street address."),
+    ])
+    city = StripWhitespaceStringField('Town or city', validators=[
+        DataRequired(message="You need to enter the town or city."),
+    ])
+    postcode = StripWhitespaceStringField('Postcode', validators=[
+        DataRequired(message="You need to enter the postcode."),
+    ])
+
+
+class EditRegisteredCountryForm(Form):
+    def __init__(self, valid_countries=None, **kwargs):
+        super().__init__(**kwargs)
+        if valid_countries:
+            self.registrationCountry.validators[1].values = [country[1] for country in valid_countries]
+
+    registrationCountry = StripWhitespaceStringField('Country', validators=[
+        DataRequired(message="You need to enter the country."),
+        AnyOf(values=[], message="This is not a valid country"),
```

Comment:
Hi @constancecerf,

We identified an edge case where user tries to manipulate html to enter an invalid country to the database. We built a validation safeguarding against that unlikely but possible scenario. And this validation needs a message informing the user what they did wrong. 

For now we used: "This is not a valid country". Let me know if this is alright or if we need to change it :).